pkgs/build-support/fetchurl/default.nix

```
@@ -104,7 +104,9 @@ let
     if urls != [] && url == "" then
       (if lib.isList urls then urls
        else throw "`urls` is not a list")
-    else if urls == [] && url != "" then [url]
+    else if urls == [] && url != "" then
+      (if lib.isString url then [url]
+       else throw "`url` is not a string")
```

Wouldn't this work as well? Or do we _really_ have to check that the string is nonempty? I guess this would fail descriptively enough when actually fetching?

```suggestion
    else if urls == [] && builtins.isString url then [url]
```

---

Empty `url` is the default.  We really want to throw something if neither is passed.

---

Oh right, definitely.